src/ng/directive/ngModelOptions.js

```diff
@@ -0,0 +1,358 @@
+'use strict';
+
+var ngModelOptionsDefaults = {
+  updateOn: 'default',
+  debounce: 0,
+  getterSetter: false,
+  allowInvalid: false,
+  timezone: null
+};
+
+
+/**
+ * @ngdoc type
+ * @name ModelOptions
+ * @description
+ * A container for the options set by the {@link ngModelOptions} directive
+ */
+function ModelOptions(options, parentOptions) {
+  this.$$parentOptions = parentOptions;
+  this.$$options = extend({}, options); // make a shallow copy
+  this.$$extendParent();
+  defaults(this.$$options, ngModelOptionsDefaults);
+  this.$$processOptions();
+}
+
+var DEFAULT_REGEXP = /(\s+|^)default(\s+|$)/;
+
+ModelOptions.prototype = {
+
+  /**
+   * @ngdoc method
+   * @name ModelOptions#getOption
+   * @param {string} name the name of the option to retrieve
+   * @returns {*} the value of the option
+   * @description
+   * Returns the value of the given option
+   */
+  getOption: function(name) {
+    return this.$$options[name];
+  },
+
+  /**
+   * @ngdoc method
+   * @name ModelOptions#createChild
+   * @param {Object} options a hash of options for the new child that will override the parent's options
+   * @return {ModelOptions} a new `ModelOptions` object initialized with the given options.
+   */
+  createChild: function(options) {
+    return new ModelOptions(options, this.$$options);
+  },
+
+  $$extendParent: function() {
+    var inheritAll = false;
+
+    // Inherit options from the parent if specified by the value `"$inherit"`
+    forEach(this.$$options, /* @this */function(option, key) {
+      if (option === '$inherit') {
+        if (key === '*') {
+          inheritAll = true;
+        } else {
+          this.$$options[key] = this.$$parentOptions[key];
+        }
+      }
+    }, this);
+
+    // We have a property of the form: `"*": "$inherit"`
+    if (inheritAll) {
+      delete this.$$options['*'];
+      defaults(this.$$options, this.$$parentOptions);
+    }
+  },
+
+  $$processOptions: function() {
+    var options = this.$$options;
+    // updateOn and updateOnDefault
+    if (isDefined(options.updateOn) && options.updateOn.trim()) {
```

Comment:
Right! Now this is a good review point.  I knew there must be something that needed fixing. I'll take a look right now.